### PR TITLE
Add python form & http accessors and http_client_interface

### LINKED ISSUE #1198: Python HTTP client & server interfaces
Port the Ruby HTTP client and HTTP server interfaces to Python

Add documentation about both to the Interface docs

### PR DESCRIPTION
**Pull request by jmthomas**
closes #1198

### CODE REVIEW

File: openc3/lib/openc3/accessors/form_accessor.rb [hunk 1]
```diff
@@ -26,16 +26,7 @@ def self.read_item(item, buffer)
...
-          if value
-            if not Array === value
-              value_temp = []
-              value_temp << value
-              value = value_temp
-            end
-            value << ary_value
-          else
-            value = ary_value
-          end
```
Comment (ryanmelt) [inline openc3/lib/openc3/accessors/form_accessor.rb:38]: Why did you remove all this?  HTML Forms can have the same item multiple times and that forms an array.

**Review by ryanmelt — CHANGES_REQUESTED** [review]

File: openc3/lib/openc3/accessors/form_accessor.rb (hunk 1 above)
Comment (jmthomas) [reply openc3/lib/openc3/accessors/form_accessor.rb:38]: I wasn't sure what this code was doing. `value` is set to nil so line 29 is always false. Didn't realize you could have multiple values with the same name but I guess that is how you handle checkbox groups.

Comment (ryanmelt) [reply openc3/lib/openc3/accessors/form_accessor.rb:38]: Line 29 is not always false if line 33 or 37 has already ran.
We only return an array if the value should be an array (typically the form contains the same name multiple times).

**Review by ryanmelt — APPROVED** [review]
